react/src/hooks/backendai.tsx

```diff
@@ -33,3 +35,137 @@ export const useResourceSlots = () => {
     },
   ] as const;
 };
+
+interface UserInfo {
+  full_name: string;
+  email: string;
+}
+
+type mutationOptions<T> = {
+  onSuccess?: (value: T) => void;
+  onError?: (error: any) => void;
+};
+
+export const useCurrentUserInfo = () => {
+  const baiClient = useSuspendedBackendaiClient();
+
+  const [userInfo, _setUserInfo] = useState<UserInfo>({
+    full_name: baiClient.full_name,
+    email: baiClient.email,
+  });
+
+  const getUsername = () => {
+    let name: string =
+      _.trim(userInfo.full_name).length > 0
+        ? userInfo.full_name
+        : userInfo.email;
+    // mask username only when the configuration is enabled
+    if (baiClient._config.maskUserInfo) {
+      const maskStartIdx = 2;
+      const emailPattern =
+        /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+      const isEmail: boolean = emailPattern.test(name);
+      const maskLength = isEmail
+        ? name.split('@')[0].length - maskStartIdx
+        : name.length - maskStartIdx;
+      name = maskString(name, '*', maskStartIdx, maskLength);
+    }
+    return name;
+  };
+
+  useEffect(() => {
+    const handler = (e: any) => {
+      if (baiClient.supports('change-user-name')) {
+        const input = e.detail;
+        _setUserInfo((v) => ({
+          ...v,
+          full_name: input,
+        }));
+      }
+    };
+    document.addEventListener('current-user-info-changed', handler);
+    return () => {
+      document.removeEventListener('current-user-info-changed', handler);
+    };
+  }, []);
+
+  const mutationToUpdateUserFullName = useTanMutation({
```

Comment:
I cannot modify the username more than once

Reply:
I removed the request condition for updateFullName in the hook. It's better to decide whether to skip the request or not where it's used.
Thank you for reporting.